apps/ledger-live-desktop/src/renderer/screens/customImage/Step1ChooseImage.tsx

```diff
@@ -1,19 +1,100 @@
-import React from "react";
+import React, { useCallback, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { NFTMetadata } from "@ledgerhq/types-live";
+import { getMetadataMediaTypes } from "~/helpers/nft";
 import { ImageBase64Data } from "~/renderer/components/CustomImage/types";
 import ImportImage from "~/renderer/components/CustomImage/ImportImage";
+import ImportNFT from "~/renderer/components/CustomImage/ImportNFT";
+import NFTGallerySelector from "~/renderer/components/CustomImage/NFTGallerySelector";
 import { StepProps } from "./types";
 import StepContainer from "./StepContainer";
+import { Box } from "@ledgerhq/react-ui";
+import StepFooter from "./StepFooter";
+import {
+  ImageLoadFromNftError,
+  ImageDownloadError,
+} from "@ledgerhq/live-common/customImage/errors";
 
 type Props = StepProps & {
   onResult: (res: ImageBase64Data) => void;
   setLoading: (_: boolean) => void;
 };
 
+const extractNftBase64 = (metadata: NFTMetadata) => {
+  const mediaTypes = metadata ? getMetadataMediaTypes(metadata) : null;
+  const mediaSizeForCustomImage = mediaTypes
+    ? ["original", "big", "preview"].find(size => mediaTypes[size] === "image")
+    : null;
+  const customImageUri =
+    (mediaSizeForCustomImage && metadata?.medias?.[mediaSizeForCustomImage]?.uri) || null;
+  return customImageUri;
+};
+
 const StepChooseImage: React.FC<Props> = props => {
   const { setLoading, onResult, onError } = props;
+  const { t } = useTranslation();
+
+  const [isShowingNftGallery, setIsShowingNftGallery] = useState<boolean>(false);
+  const [currentBase64, setCurrentBase64] = useState<ImageBase64Data | null>(null);
+
+  const handleClickNext = useCallback(() => {
+    if (!currentBase64) {
+      onError(new ImageLoadFromNftError());
+    } else onResult(currentBase64);
+  }, [currentBase64, onError, onResult]);
+
+  const handleClickPrevious = useCallback(() => {
+    setCurrentBase64(null);
+    setIsShowingNftGallery(false);
+  }, []);
+
+  const handlePickNft = useCallback(
+    (nftMetadata: NFTMetadata) => {
+      const uri = extractNftBase64(nftMetadata);
```

Comment:
You can use the function `urlContentToDataUri()` from Olivier in `src/renderer/components/CustomImage/shared.ts` to save some logic here